pyro/util.py

```diff
@@ -30,20 +30,34 @@ def set_rng_seed(rng_seed):
 
 def torch_isnan(x):
     """
-    A convenient function to check if a Tensor contains all nan; also works with numbers
+    A convenient function to check if a Tensor contains any nan; also works with numbers
     """
     if isinstance(x, numbers.Number):
         return x != x
-    return torch.isnan(x).all()
+    return torch.isnan(x).any()
 
 
 def torch_isinf(x):
     """
-    A convenient function to check if a Tensor contains all inf; also works with numbers
+    A convenient function to check if a Tensor contains any inf; also works with numbers
     """
     if isinstance(x, numbers.Number):
         return x == float('inf')
-    return (x == float('inf')).all()
+    return (x == float('inf')).any()
+
+
+def warn_if_nan(name, value):
```

Comment:
If we have use cases where it is fine to have `+inf` values but not `NaN`s, then it would make sense to have an optional kwarg `warn_if_posinf=True`, so that optionally turn off the check on `+inf` values. That said, if you haven't found any code / failing tests yet, we can address this later.

Reply:
What do you think about changing `name` to an optional but more general `msg=""` kwarg, which could be specified as `msg="log_prob_sum at site {}".format(name)`, or `msg="site: {}".format(name)`, and used as follows below:

```
warnings.warn("Encountered inf. " + msg)
```